### DISCUSSION TITLE
Search - General Questions

### DISCUSSION THREAD

**Discussion by shaneholloman** [squidfunk/mkdocs-material#2861] [Q&A] [answered]
## Question 1 - Search "Fields"
Lunr has the notion of:

```sh
idx.search('title:foo* bar')
```

In  mkdocs-material' case, this works well, awesome

However, I couldn't find any other "fields" that mkdocs-material is currently leveraging.  
May we get a list of fields we can explicitly search within?

## Question 2 - Searching Fenced Code

For instance:  

```sh
sudo anything here
```

Is not visible in search. This is where Question 1 comes to play

For example I would like to search a command I forgot like

```bash
sudo -H pip install --ignore-installed six --upgrade six
```

By using a search like

```sh
code:sudo -H pip install *
```

Again this works great if I know the available fields.  
So for titles:

```sh
title:sudo -H pip install
```

Is great, but only if that was in a title (h1, h2, h3 etc.), which it sometimes is, but normally isn't.

**Comment by squidfunk** [comment, marked as answer]
### Question 1 - Search "Fields"

Searchable fields are:

- `title`
- `text`
- `tags` (currently Insiders only)

Note that the syntax demands for prefixing _every_ word with the field, i.e. `title:foo title:bar`, because `title:foo bar`means "documents with `foo` in `title` and `bar` in any field". This will, however, yield an `OR` search. If you want an `AND` search, you would have to search like this:

```
+title:foo +title:bar
```

... or let Material for MkDocs [preprocess the query string](https://squidfunk.github.io/mkdocs-material/setup/setting-up-site-search/#query-transformation) and use quotes as known from Google:

```
"title:foo title:bar"
```

Why is this not documented? I think you very much see how complex and ugly the syntax is and most users just don't bother. My goal is to provide a search implementation that returns good results with simple keyword search and presents them in a way so that you can quickly skim through the results, thus omitting the need for explanation.

### Question 2 - Searching Fenced Code

Code blocks are just converted to regular text when MkDocs is building the index, so there're no remnants of "hey, this is text inside a code block". One would have to monkey-patch the search plugin to somehow set markers and allow for dedicated indexing of code examples. If you look into `search/search_index.json`, which is what MkDocs generates, you'll see that we have no information your feature request could be based on.
